pytestqt/wait_signal.py

```diff
@@ -123,6 +122,9 @@ def _quit_loop_by_signal(self, *args):
         """
         quits the event loop and marks that we finished because of a signal.
         """
+        if self.callback:
+            if not self.callback(*args):
+                return  # parameter check did not pass
```

Comment:
I wonder if we should store the received args, so we can provide a better error message? 

Consider:

```python
def is_disconnected(text):
    return text == 'END'

with qtbot.waitSignal(conn.text_received, callback=is_disconnected):
    conn.request_disconnect()   
```

`conn.text_received` is emitted with `['WAIT_CLIENTS', 'PROCESSING']` and eventually times out.

In this situation all the user will see is a `TimeoutError` and won't have any hints at what might have gone wrong. I suppose the user would then have to add `print` statements at `is_disconnected` to try to figure out the problem.

I think if we store the received arguments when there is a callback, we can reuse that later in the exception message:

```
TimeoutError:  `text_received` emitted with ['WAIT_CLIENTS', 'PROCESSING'] 
but did not satisfy the `is_disconnected` callback.
```

That idea also implies that the same arguments should be available for inspection later in case `raising` is `False`, probably as a list of `*args`: 

```python
blocker.all_args == ['WAIT_CLIENTS', 'PROCESSING']
```